tag.php

```diff
@@ -14,32 +15,44 @@
 		'layout'        => 'vertical',
 		'layout__type'  => 'primary'
 	);
+
+	if( class_exists( 'UCF_Events_Config' ) ) {
+		$events_feed_url = UCF_Events_Config::get_option_or_default( 'feed_url' );
+		$all_events_link = str_replace( 'feed.json', '', $events_feed_url );
+	}
 ?>
 
 <div class="container mt-4 mb-5 pb-sm-4">
 	<div class="row">
 		<div class="col-md-8">
-			<?php
-				if( isset( $posts ) && isset( $first_post ) ) {
-					echo today_post_list_display_feature( '', $first_post, $atts );
-
-					$atts['posts_per_row'] = 3;
-					echo today_post_list_display_feature( '', $posts, $atts );
-				} else {
-					echo "No results found.";
-				}
-			?>
+			<?php if ( $first_post ): ?>
+				<?php echo today_display_feature_vertical( $first_post, array( 'layout__type' => 'primary' ) ); ?>
+			<?php endif; ?>
+
+			<?php if ( $posts ): ?>
+				<div class="row">
+					<?php foreach ( $posts as $post ): ?>
+						<div class="col-lg-4">
+							<?php echo today_display_feature_vertical( $post ); ?>
+						</div>
+					<?php endforeach; ?>
+				</div>
+			<?php endif; ?>
+
+			<?php if ( ! $first_post && ! $posts ): ?>
+				<p>No results found.</p>
+			<?php endif; ?>
 		</div>
 		<div class="col-md-4">
-			<h2 class="heading-underline h6">Events at UCF</h2>
-			<?php echo do_shortcode('[ucf-events feed_url="https://events.ucf.edu/upcoming/feed.json" layout="classic" offset="1" limit="4" title=""]'); ?>
-			<a href="https://events.ucf.edu/upcoming/">View All Events</a>
+			<h2 class="h6 text-uppercase text-default-aw mb-4">Events at UCF</h2>
+			<?php echo do_shortcode('[ucf-events feed_url="' . $events_feed_url . '" layout="classic" offset="1" limit="4" title=""]'); ?>
+			<a href="<?php echo $all_events_link; ?>">View All Events</a>
 
-			<h2 class="heading-underline h6 mt-5">UCF In the News</h2>
+			<h2 class="h6 text-uppercase text-default-aw mb-4 mt-5">UCF In the News</h2>
 			<?php echo do_shortcode('[ucf-post-list layout="condensed" post_type="ucf_resource_link" numberposts="4"]'); ?>
-			<a href="in-the-news/">View All</a>
+			<a href="<?php echo get_permalink( get_page_by_title( "UCF in the News" ) ); ?>">View All</a>
```

Comment:
Consider adding a check to ensure a valid page was found + a permalink was retrieved before printing this `<a>` tag